File: leetcode/questions/713/java/713.java
class Solution {
    public int numSubarrayProductLessThanK(int[] nums, int k) {
        int left = 0;
        int ans = 0;
        int prod = 1;

        for (int right = 0; right < nums.length; ++right) {
            prod *= nums[right];

            while (left <= right && prod >= k) {
                prod /= nums[left++];
            }

            ans += (right - left + 1);
        }

        return ans;
    }
}

File: leetcode/questions/713/python/713.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        ans = 0
        prod = 1
        left = 0

        for right, num in enumerate(nums):
            prod *= num

            while left <= right and prod >= k:
                prod //= nums[left]
                left += 1

            ans += (right - left + 1)

        return ans